lib/internal/preferences/FilePreferenceStore.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © 2019 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { HandlerContext } from "@atomist/automation-client";
+import { PreferenceStoreFactory } from "@atomist/sdm";
+import * as fs from "fs-extra";
+import * as os from "os";
+import * as path from "path";
+import {
+    lock,
+    unlock,
+} from "proper-lockfile";
+import {
+    AbstractPreferenceStore,
+    Preference,
+} from "./AbstractPreferenceStore";
+
+interface PreferenceFile {
+    [key: string]: { value: string, ttl?: number };
+}
+
+/**
+ * Factory to create a new FilePreferenceStore instance
+ * @param ctx
+ * @constructor
+ */
+export const FilePreferenceStoreFactory: PreferenceStoreFactory = ctx => new FilePreferenceStore(ctx);
+
+/**
+ * PreferenceStore implementation that stores preferences in a shared file.
+ * Note: this implementation attempts to lock the preference file before reading or writing to it
+ * but it is not intended for production usage.
+ */
+export class FilePreferenceStore extends AbstractPreferenceStore {
+
+    constructor(private readonly context: HandlerContext,
+                private readonly fileName: string = "client.prefs.json") {
+        super(context);
+    }
+
+    public path(): string {
+        const p = path.join(os.userInfo().homedir, ".atomist", "prefs");
```

Comment:
Any reason you didn't just use `os.homedir()` https://nodejs.org/api/os.html#os_os_homedir ?